components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchProducer.java

```diff
@@ -44,8 +46,31 @@ public SpringBatchProducer(SpringBatchEndpoint endpoint, JobLauncher jobLauncher
 
     @Override
     public void process(Exchange exchange) throws Exception {
+
         JobParameters jobParameters = prepareJobParameters(exchange.getIn().getHeaders());
-        JobExecution jobExecution = jobLauncher.run(job, jobParameters);
+        String messageJobName = jobParameters.getString(SpringBatchComponent.DYNAMIC_JOBNAME);
+
+        Job job2run = this.job;
+
+        if (messageJobName != null) {
+            Job dynamicJob = CamelContextHelper.mandatoryLookup(getEndpoint().getCamelContext(), messageJobName, Job.class);
+
+            job2run = dynamicJob;
+
+            if (job2run == null) {
```

Comment:
This check is redundant, `CamelContextHelper.mandatoryLookup` will throw `NoSuchBeanException` if it cannot find the bean.